tools/infrasys/src/s3.rs

```diff
@@ -176,8 +179,11 @@ pub async fn upload_file(
     prefix: &str,
     file_path: &Path,
 ) -> Result<()> {
-    let s3_client =
-        S3Client::new(Region::from_str(region).context(error::ParseRegionSnafu { what: region })?);
+    let config = aws_config::from_env()
```

Comment:
Is it possible/desirable to pass the `aws_config` instead of recreating it every time?

Reply:
I'm hesitant to add the config, mostly because the most obvious place to add it keeps track of lifetimes, while the aws sdk wants a static lifetime for everything. It makes things a bit awkward, as you can see from the Region changes.